deeprankcore/dataset.py

```diff
@@ -658,17 +658,40 @@ def __init__( # pylint: disable=too-many-arguments, too-many-locals
         if not train:
             if not isinstance(dataset_train, GraphDataset):
                 raise TypeError("Please provide a valid training GraphDataset.")
-
-            if dataset_train.features_transform is not None:
-                self.features_transform = dataset_train.features_transform
-            else:
-                self.features_transform = None
+            
+            #set features_transform as the same in dataset_train
+            self.features_transform = dataset_train.features_transform
```

Comment:
The if statement should stay, since it may be that `dataset_train.features_transform` is None. In that case, we want `self.features_transform` to be None as well.